File: snippets/kotlin/src/main/kotlin/Gossip.kt
package rust.nostr.snippets

import kotlinx.coroutines.runBlocking
// ANCHOR: gossip
import org.nostrdevkit.sdk.*

suspend fun buildGossipClient(databasePath: String): Client {
    val gossip = NostrGossip.sqlite(databasePath)
    val client = ClientBuilder().gossip(gossip).build()
    val capabilities = RelayCapabilities.discovery()

    client.addRelay(RelayUrl.parse("wss://relay.damus.io"), capabilities)
    client.addRelay(RelayUrl.parse("wss://purplepag.es"), capabilities)
    client.connect()

    return client
}
// ANCHOR_END: gossip

fun gossipMain() = runBlocking {
    buildGossipClient("./data/gossip.sqlite")
}

File: snippets/csharp/Gossip.cs
// ANCHOR: gossip
namespace Snippets;

using Nostr.Sdk;

public static class GossipExample
{
    public static async Task<Client> BuildGossipClient(string databasePath)
    {
        var gossip = await NostrGossip.Sqlite(databasePath);
        var client = new ClientBuilder().Gossip(gossip).Build();
        var capabilities = RelayCapabilities.Discovery();

        await client.AddRelay(RelayUrl.Parse("wss://relay.damus.io"), capabilities);
        await client.AddRelay(RelayUrl.Parse("wss://purplepag.es"), capabilities);
        await client.Connect();

        return client;
    }
}
// ANCHOR_END: gossip
